service/src/test/java/bio/terra/cbas/controllers/TestMethodsApiController.java

```diff
@@ -844,6 +844,43 @@ void dontStoreMethodDetailsForDockstoreMethod() throws Exception {
     assertNull(githubMethodDetailsDao.getMethodSourceDetails(postMethodResponse.getMethodId()));
   }
 
+  @Test
+  void postMethodForUserWithNoToken() throws Exception {
```

Comment:
It might be good to add the error cases also as part of unit testing - especially the ones where GithubService.isPrivateRepo throws an error or EcmService throws an error